modules/gmc-agent/src/main/java/org/gridgain/utils/QueryUtils.java

```diff
@@ -371,4 +382,104 @@ public static String binaryToString(BinaryObject obj) {
                 "hash", hash, false,
                 "typeId", obj.type().typeId(), true);
     }
+
+    /**
+     * @param cacheName Cache name.
+     * @param types Types.
+     * @return Cache sql metadata
+     */
+    public static CacheSqlMetadata queryTypesToMetadata(String cacheName, Collection<GridQueryTypeDescriptor> types) {
+        CacheSqlMetadata metadata = new CacheSqlMetadata().setCacheName(cacheName);
+
+        for (GridQueryTypeDescriptor type : types) {
```

Comment:
Please add comment that this code was copy-pasted from SQL internals.

Reply:
Done.